sdk/provisioning/Azure.Provisioning/src/ProvisioningPlan.Deploy.cs

```diff
@@ -0,0 +1,237 @@
+        string deploymentName = $"{resourceGroupName}-deployment";
+        Azure.ResourceManager.Resources.Models.ArmDeploymentContent deploymentContent =
+            new(
+                new Azure.ResourceManager.Resources.Models.ArmDeploymentProperties(Azure.ResourceManager.Resources.Models.ArmDeploymentMode.Incremental)
+                {
+                    Template = BinaryData.FromString(armTemplate)
+                });
+        try
+        {
+            ArmOperation<ArmDeploymentResource> deploymentCreation = async ?
+                await deployments.CreateOrUpdateAsync(WaitUntil.Completed, deploymentName, deploymentContent, cancellationToken).ConfigureAwait(false) :
+                deployments.CreateOrUpdate(WaitUntil.Completed, deploymentName, deploymentContent, cancellationToken);
+            return deploymentCreation.Value;
+        }
+        catch (RequestFailedException ex) when (ex.ErrorCode == "DeploymentFailed")
+        {
+            // If something goes wrong, fetch the deployment info which will
+            // contain the error, etc.
+            return async ?
+                await deployments.GetAsync(deploymentName, cancellationToken).ConfigureAwait(false) :
+                deployments.Get(deploymentName, cancellationToken);
+        }
+    }
+
+    private ProvisioningDeployment ProcessDeploymentInternal(ArmDeploymentResource deployment)
+    {
+        // Pull outputs from the results that are structured like
+        // {"name":{"type":"String","value":"xyz"}}
+        Dictionary<string, object?> outputs = [];
+        if (deployment.Data.Properties.Outputs is not null)
+        {
+            JsonElement doc = JsonDocument.Parse(deployment.Data.Properties.Outputs.ToMemory()).RootElement;
+            foreach (JsonProperty prop in doc.EnumerateObject())
+            {
+                try
+                {
+                    JsonElement value = prop.Value.GetProperty("value");
+                    // TODO: Should we just copy the source here so we don't need to rewrap it in BinaryData?
+                    // Use Azure.Core's ToObjectFromJson that does all the reasonable
+                    // things and also turns objects into Dictionary<string, object> and
+                    // arrays into List<object>
+                    object? parsed = BinaryData.FromString(value.GetRawText()).ToObjectFromJson();
+                    outputs[prop.Name] = parsed;
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException($"Failed to parse output {prop}", e);
+                }
+            }
+        }
+
+        // Patch up output references
+        foreach (BicepOutput output in _infrastructure.GetResources().OfType<BicepOutput>())
+        {
+            if (outputs.TryGetValue(output.ResourceName, out object? value) &&
+                value is not null)
+            {
+                output.Value = value;
+            }
+        }
```

Comment:
And even just update all the live output instances so you can `Console.WriteLine(acctName)` since you already had a variable for it.